src/sigma/test/serialize_test.cpp

```diff
@@ -21,21 +21,10 @@ BOOST_AUTO_TEST_CASE(group_element_serialize)
 
 BOOST_AUTO_TEST_CASE(group_element_invalid)
 {
-    secp_primitives::GroupElement initial;
-    initial.randomize();
-    unsigned char buffer [initial.memoryRequired()];
-    initial.serialize(buffer);
-    //Shift all elements to right by 2, to make it invalid
-    for (int i = initial.memoryRequired() - 1; i > 1; i--)
-        buffer[i] = buffer[i - 2];
-    buffer[0] = std::rand();
-    buffer[1] = std::rand();
-
-    // Making the point not infinity as it is allowed in deserialization
-    buffer[33] = 0;
-
+    // Invalid GroupElement generated in advance
+    std::string buffer = "956f89850d79edd157360bb8608b8a336ca6e00dc057b4182540db5536745ef57b00";
```

Comment:
Can you provide an explanation for how this particular invalid serialization was derived, and why it is suitable to test the desired functionality?

Reply:
Previous test code sometimes failed there is possibility to generate valid point from such semi-random string, I generated an invalid point checked that, and put it here.

Reply:
Sure, but what makes it a good test candidate? I'm of the opinion that magic numbers should be easily explainable or reproducible where possible.